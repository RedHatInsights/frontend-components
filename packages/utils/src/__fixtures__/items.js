export const severityLevels = ['high', 'medium', 'low', 'unknown'];

export default (count) => {
  let currentCount = 0;
  return [...new Array(count > 0 ? count : 1)]
    .map(() => {
      currentCount++;
      return {
        id: currentCount,
        severity: severityLevels[currentCount % 4],
        name: 'TEST ITEM #' + currentCount,
        description: 'DESCRIPTION',
      };
    })
    .sort();
};
