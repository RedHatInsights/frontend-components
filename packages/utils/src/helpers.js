import React from 'react';
import merge from 'lodash/merge';
import mapKeys from 'lodash/mapKeys';
import ContentLoader from 'react-content-loader';
import { Tooltip } from '@patternfly/react-core';

export const CSV_TYPE = 'text/csv;charset=utf-8;';
export const JSON_TYPE = 'data:text/json;charset=utf-8,';
const monthMap = [
    'Jan', 'Feb', 'Mar', 'Apr', 'May', 'June', 'July', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec'
];

export function mergeArraysByKey(arrays, key = 'id') {
    let mergedObject = merge(...arrays.map(row => mapKeys(row, a => a && a[key])));
    return Object.values(mergedObject);
}

export function downloadFile(data, filename = `${new Date().toISOString()}`, format = CSV_TYPE) {
    const type = format === 'json' ? JSON_TYPE : CSV_TYPE;
    const blob = new Blob([ data ], { type });
    const link = document.createElement('a');
    link.setAttribute('href', URL.createObjectURL(blob));
    link.setAttribute('download', `${filename}.${format}`);
    link.style.visibility = 'hidden';
    document.body.appendChild(link);
    link.click();
    document.body.removeChild(link);
}

export function parseCvssScore(cvssV2, cvssV3, withLabels = false) {
    const v2Tooltip = 'Prior to 2016 (approximately), CVEs were scored with Common Vulnerability Scoring System v2.';
    const naTooltip = 'CVEs published before 2005 (approximately) did not have a CVSS Base Score.';
    return (
        (cvssV3 && parseFloat(cvssV3).toFixed(1)) ||
        (cvssV2 && (
            <Tooltip content={ v2Tooltip } position={ 'left' }>
                <span>{ `${parseFloat(cvssV2).toFixed(1)}` } { withLabels && '(CVSSv2)' }</span>
            </Tooltip>
        )) || (
            <Tooltip content={ naTooltip } position={ 'left' }>
                <span>N/A</span>
            </Tooltip>
        )
    );
}

export function processDate(dateString) {
    const date = new Date(dateString);
    const month = monthMap[date.getMonth()];
    const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate();
    if (!month || isNaN(day)) {
        return 'N/A';
    }

    return `${day} ${month} ${date.getFullYear()}`;
}

export const RowLoader = props => (
    <ContentLoader height={ 20 } width={ 480 } { ...props }>
        <rect x="30" y="0" rx="3" ry="3" width="250" height="7" />
        <rect x="300" y="0" rx="3" ry="3" width="70" height="7" />
        <rect x="385" y="0" rx="3" ry="3" width="95" height="7" />
        <rect x="50" y="12" rx="3" ry="3" width="80" height="7" />
        <rect x="150" y="12" rx="3" ry="3" width="200" height="7" />
        <rect x="360" y="12" rx="3" ry="3" width="120" height="7" />
        <rect x="0" y="0" rx="0" ry="0" width="20" height="20" />
    </ContentLoader>
);

export function getBaseName(pathname, level = 2) {
    let release = '/';
    const pathName = pathname.replace(/(#|\?).*/, '').split('/');

    pathName.shift();

    if (pathName[0] === 'beta') {
        pathName.shift();
        release = `/beta/`;
    }

    return [ ...new Array(level) ].reduce((acc, _curr, key) => {
        return `${acc}${pathName[key] || ''}${key < (level - 1) ? '/' : ''}`;
    }, release);
}
