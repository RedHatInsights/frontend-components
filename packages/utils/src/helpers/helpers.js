import merge from 'lodash/merge';
import mapKeys from 'lodash/mapKeys';

export const CSV_TYPE = 'text/csv;charset=utf-8;';
export const JSON_TYPE = 'data:text/json;charset=utf-8,';
const monthMap = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'June', 'July', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec'];

export function mergeArraysByKey(arrays, key = 'id') {
  let mergedObject = merge(...arrays.map((row) => mapKeys(row, (a) => a && a[key])));
  return Object.values(mergedObject);
}

export function downloadFile(data, filename = `${new Date().toISOString()}`, format = CSV_TYPE) {
  const type = format === 'json' ? JSON_TYPE : CSV_TYPE;
  const blob = new Blob([data], { type });
  const link = document.createElement('a');
  link.setAttribute('href', URL.createObjectURL(blob));
  link.setAttribute('download', `${filename}.${format}`);
  link.style.visibility = 'hidden';
  document.body.appendChild(link);
  link.click();
  document.body.removeChild(link);
}

export function processDate(dateString) {
  const date = new Date(dateString);
  const month = monthMap[date.getMonth()];
  const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate();
  if (!month || isNaN(day)) {
    return 'N/A';
  }

  return `${day} ${month} ${date.getFullYear()}`;
}

export function getBaseName(pathname, level = 2) {
  let release = '/';
  const pathName = pathname.replace(/(#|\?).*/, '').split('/');

  pathName.shift();

  if (pathName[0] === 'beta') {
    pathName.shift();
    release = `/beta/`;
  }

  return [...new Array(level)].reduce((acc, _curr, key) => {
    return `${acc}${pathName[key] || ''}${key < level - 1 ? '/' : ''}`;
  }, release);
}

export const generateFilter = (data, path = 'filter', options) =>
  Object.entries(data || {}).reduce((acc, [key, value]) => {
    const newPath = `${path || ''}[${key}]${Array.isArray(value) ? `${options?.arrayEnhancer ? `[${options.arrayEnhancer}]` : ''}[]` : ''}`;
    if (value instanceof Function || value instanceof Date) {
      return acc;
    }

    return {
      ...acc,
      ...(Array.isArray(value) || typeof value !== 'object' ? { [newPath]: value } : generateFilter(value, newPath, options)),
    };
  }, {});

export const toUpperCase = (text = '') => `${text.charAt(0).toUpperCase()}${text.slice(1)}`;
