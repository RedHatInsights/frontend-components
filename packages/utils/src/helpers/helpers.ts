import merge from 'lodash/merge';
import mapKeys from 'lodash/mapKeys';

export const CSV_TYPE = 'text/csv;charset=utf-8;';
export const JSON_TYPE = 'data:text/json;charset=utf-8,';
const monthMap = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'June', 'July', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec'];

export function mergeArraysByKey(arrays: Record<string, unknown>[][], key = 'id') {
  // @ts-ignore
  const mergedObject = merge(...arrays.map((row) => mapKeys(row, (a) => a && a[key])));
  return Object.values(mergedObject);
}

export function downloadFile(data: unknown, filename = `${new Date().toISOString()}`, format = 'csv') {
  const type = format === 'json' ? JSON_TYPE : CSV_TYPE;
  const blob = new Blob([data as unknown as BlobPart], { type });
  const link = document.createElement('a');
  link.setAttribute('href', URL.createObjectURL(blob));
  link.setAttribute('download', `${filename}.${format}`);
  link.style.visibility = 'hidden';
  document.body.appendChild(link);
  link.click();
  document.body.removeChild(link);
}

export function processDate(dateString: string | number | Date) {
  const date = new Date(dateString);
  const month = monthMap[date.getMonth()];
  const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate();
  if (!month || isNaN(day as number)) {
    return 'N/A';
  }

  return `${day} ${month} ${date.getFullYear()}`;
}

export function getBaseName(pathname: string, level = 2) {
  const previewFragment = pathname.split('/')[1];
  const isPreview = ['beta', 'preview'].includes(previewFragment);
  let release = '/';
  const pathName = pathname.replace(/(#|\?).*/, '').split('/');

  pathName.shift();

  if (isPreview) {
    pathName.shift();
    release = `/${previewFragment}/`;
  }

  return [...new Array(level)].reduce<string>((acc, _curr, key) => {
    return `${acc}${pathName[key] || ''}${key < level - 1 ? '/' : ''}`;
  }, release);
}

export type FilterData<T = Record<string, unknown | string | Date | ((...args: unknown[]) => unknown)>> =
  | object
  | null
  | Record<string, unknown | string | Date | ((...args: unknown[]) => unknown) | T>;

export const generateFilter = (data: FilterData, path = 'filter', options?: { arrayEnhancer: string }): Record<string, string> =>
  Object.entries(data || {}).reduce((acc, [key, value]) => {
    const newPath = `${path || ''}[${key}]${Array.isArray(value) ? `${options?.arrayEnhancer ? `[${options.arrayEnhancer}]` : ''}[]` : ''}`;
    if (value instanceof Function || value instanceof Date) {
      return acc;
    }

    return {
      ...acc,
      ...(Array.isArray(value) || typeof value !== 'object' ? { [newPath]: value } : generateFilter(value, newPath, options)),
    };
  }, {});

export const toUpperCase = (text = '') => `${text.charAt(0).toUpperCase()}${text.slice(1)}`;
