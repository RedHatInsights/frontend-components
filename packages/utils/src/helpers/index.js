export * as default from './helpers';
export * from './helpers';
