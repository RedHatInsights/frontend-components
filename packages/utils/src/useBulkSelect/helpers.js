export const compileTitle = (itemsTotal, titleOption) => {
  if (typeof titleOption === 'string') {
    return titleOption;
  } else if (typeof titleOption === 'function') {
    return titleOption(itemsTotal);
  } else {
    return `${itemsTotal} selected`;
  }
};

export const checkboxState = (selectedItemsTotal, itemsTotal) => {
  if (selectedItemsTotal === 0) {
    return false;
  } else if (selectedItemsTotal === itemsTotal) {
    return true;
  } else {
    return null;
  }
};

export const selectOrUnselect = (selected) => (selected ? 'Unselect' : 'Select');

const allItemsIncluded = (items, selection = []) => items.filter((item) => selection.includes(item)).length === items.length;

export const checkCurrentPageSelected = (items, selectedItems) => {
  if (selectedItems.length === 0) {
    return false;
  } else {
    return allItemsIncluded(items, selectedItems);
  }
};

export const itemIds = (items) => items.map(({ itemId }) => itemId);

export const selectItemTransformer = (item, selectedIds = []) => ({
  ...item,
  rowProps: {
    selected: selectedIds.includes(item.itemId),
  },
});
