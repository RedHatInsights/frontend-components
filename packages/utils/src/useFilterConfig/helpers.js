import isArray from 'lodash/isEmpty';
import isObject from 'lodash/isObject';

export const stringToId = (string) => string.split(/\s+/).join('-').toLowerCase();

export const findWithString = (value) => (item) => String(item.value) === String(value);

export const defaultPlaceholder = (label) => `Filter by ${label.toLowerCase()}`;

export const defaultOnChange = (handler, label) => ({
  onChange: (_event, selectedValues, selectedValue) => {
    handler(label, selectedValue, selectedValues);
  },
});

export const flattenConfigItems = (configItem) =>
  configItem.items.flatMap((parentItem) => [parentItem, ...parentItem.items.map((item) => ({ ...item, parent: parentItem }))]);

export const configItemItemByLabel = (configItem, label) => configItem.items.find(({ label: itemLabel }) => itemLabel === label);

export const itemForValueInGroups = (configItem, value) => {
  const flatItems = flattenConfigItems(configItem);
  const item = flatItems.find(({ value: itemValue }) => {
    return `${itemValue}` === `${value}`;
  });
  return item;
};

export const itemForLabelInGroups = (configItem, label) => {
  const flatItems = flattenConfigItems(configItem);
  const item = flatItems.find(({ label: ItemLabel }) => {
    return `${ItemLabel}` === `${label}`;
  });
  return item;
};

export const isNotEmpty = (value) => (isArray(value) && value?.length > 0) || value !== '' || (isObject(value) && Object.keys(value).length > 0);
