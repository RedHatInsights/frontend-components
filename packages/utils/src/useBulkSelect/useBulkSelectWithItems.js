import { useEffect, useMemo } from 'react';
import { filterSelected } from '../useTableTools/helpers';
import { itemIds, selectItemTransformer } from './helpers';
import useBulkSelect from './useBulkSelect';

/**
 * Provides properties for a Pattternfly (based) Table and Toolbar component to implement bulk selection
 *
 * @param {Function} [onSelect] function to call when a selection is made
 * @param {Array} [items] Array of items selected when initialising
 *
 */
const useBulkSelectWithItems = ({ onSelect, items: propItems, filter, paginator, preselected, setPage }) => {
  const enableBulkSelect = !!onSelect;
  const items = propItems.map((item) => selectItemTransformer(item, preselected));
  const total = items.length;

  const filteredItems = filter?.(items) ?? items;
  const filteredTotal = filteredItems.length;
  const filtered = filteredTotal !== total;

  const paginatedItems = paginator?.(filteredItems) ?? filteredItems;
  const paginatedTotal = paginatedItems.length;

  const allCount = filteredTotal ?? total;

  const setPageMemo = useMemo(() => setPage, []);

  useEffect(() => {
    if (paginatedTotal === 0) {
      setPageMemo(-1);
    }
  }, [paginatedTotal, setPageMemo]);

  const { selectNone, selectedIds, ...bulkSelect } = useBulkSelect({
    total: allCount,
    onSelect,
    preselected,
    itemIdsInTable: async () => (filtered ? itemIds(filteredItems) : itemIds(items)),
    itemIdsOnPage: paginatedItems.map(({ itemId }) => itemId),
    identifier: 'itemId',
  });

  return enableBulkSelect
    ? {
        transformer: (item) => selectItemTransformer(item, selectedIds),
        selectedItems: filterSelected(items, selectedIds),
        selected: selectedIds,
        clearSelection: selectNone,
        ...bulkSelect,
      }
    : {};
};

export default useBulkSelectWithItems;
