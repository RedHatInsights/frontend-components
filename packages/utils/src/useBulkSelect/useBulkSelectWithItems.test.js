import { renderHook, act } from '@testing-library/react-hooks';
import { identifyItems } from '../helpers';
import items from '../__fixtures__/items';
import useBulkSelectWithItems from './useBulkSelectWithItems';

describe('useBulkSelectWithItems', () => {
  const exampleItems = identifyItems(items(20));
  const defaultOptions = {
    onSelect: () => ({}),
    items: exampleItems,
    perPage: 10,
    preselected: [],
  };
  const getBulkSelect = (result) => result.current.toolbarProps.bulkSelect;
  const getSelectNone = (result) => result.current.toolbarProps.bulkSelect.items[0];
  const getSelectPage = (result) => result.current.toolbarProps.bulkSelect.items[1];
  const getSelectAll = (result) => result.current.toolbarProps.bulkSelect.items[2];

  it('returns a bulk select configuration', () => {
    const { result } = renderHook(() => useBulkSelectWithItems(defaultOptions));

    expect(result).toMatchSnapshot();
  });

  it('returns a allows to select one', async () => {
    const item = exampleItems[5];
    const { result } = renderHook(() => useBulkSelectWithItems(defaultOptions));

    act(() => {
      result.current.tableProps.onSelect(undefined, true, 'itemId', item);
    });

    expect(result.current.transformer(item)).toMatchSnapshot();
  });

  it('returns a allows to select/deselect all', async () => {
    const { result, waitForNextUpdate } = renderHook(() => useBulkSelectWithItems(defaultOptions));
    expect(getBulkSelect(result)).toMatchSnapshot();

    act(() => {
      getSelectAll(result).onClick();
    });

    await waitForNextUpdate();
    expect(getBulkSelect(result)).toMatchSnapshot();

    act(() => {
      getSelectAll(result).onClick();
    });
    await waitForNextUpdate();
    expect(getBulkSelect(result)).toMatchSnapshot();
  });

  it('returns a allows to select/deselect page', async () => {
    const { result } = renderHook(() => useBulkSelectWithItems(defaultOptions));
    expect(getBulkSelect(result)).toMatchSnapshot();

    act(() => {
      getSelectPage(result).onClick();
    });

    expect(getBulkSelect(result)).toMatchSnapshot();

    act(() => {
      getSelectPage(result).onClick();
    });

    expect(getBulkSelect(result)).toMatchSnapshot();
  });

  it('returns respects filtered results', async () => {
    const { result, waitForNextUpdate } = renderHook(() =>
      useBulkSelectWithItems({
        ...defaultOptions,
        filter: (items) => items.slice(5, 10),
      })
    );
    expect(getBulkSelect(result)).toMatchSnapshot();

    act(() => {
      getSelectAll(result).onClick();
    });

    await waitForNextUpdate();
    expect(getBulkSelect(result)).toMatchSnapshot();
  });
});
