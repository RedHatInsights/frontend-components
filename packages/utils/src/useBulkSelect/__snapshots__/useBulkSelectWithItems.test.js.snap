// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useBulkSelectWithItems returns a allows to select one 1`] = `
Object {
  "description": "DESCRIPTION",
  "id": 6,
  "itemId": 6,
  "name": "TEST ITEM #6",
  "rowProps": Object {
    "selected": true,
  },
  "severity": "low",
}
`;

exports[`useBulkSelectWithItems returns a allows to select/deselect all 1`] = `
Object {
  "checked": false,
  "isDisabled": false,
  "items": Array [
    Object {
      "onClick": [Function],
      "props": Object {
        "isDisabled": true,
      },
      "title": "Select none",
    },
    Object {
      "onClick": [Function],
      "title": "Select page (20 items)",
    },
    Object {
      "onClick": [Function],
      "title": "Select all (20 items)",
    },
  ],
  "onSelect": [Function],
  "toggleProps": Object {
    "children": Array [
      "0 selected",
    ],
  },
}
`;

exports[`useBulkSelectWithItems returns a allows to select/deselect all 2`] = `
Object {
  "checked": true,
  "isDisabled": false,
  "items": Array [
    Object {
      "onClick": [Function],
      "props": Object {
        "isDisabled": false,
      },
      "title": "Select none",
    },
    Object {
      "onClick": [Function],
      "title": "Unselect page (20 items)",
    },
    Object {
      "onClick": [Function],
      "title": "Unselect all (20 items)",
    },
  ],
  "onSelect": [Function],
  "toggleProps": Object {
    "children": Array [
      "20 selected",
    ],
  },
}
`;

exports[`useBulkSelectWithItems returns a allows to select/deselect all 3`] = `
Object {
  "checked": false,
  "isDisabled": false,
  "items": Array [
    Object {
      "onClick": [Function],
      "props": Object {
        "isDisabled": true,
      },
      "title": "Select none",
    },
    Object {
      "onClick": [Function],
      "title": "Select page (20 items)",
    },
    Object {
      "onClick": [Function],
      "title": "Select all (20 items)",
    },
  ],
  "onSelect": [Function],
  "toggleProps": Object {
    "children": Array [
      "0 selected",
    ],
  },
}
`;

exports[`useBulkSelectWithItems returns a allows to select/deselect page 1`] = `
Object {
  "checked": false,
  "isDisabled": false,
  "items": Array [
    Object {
      "onClick": [Function],
      "props": Object {
        "isDisabled": true,
      },
      "title": "Select none",
    },
    Object {
      "onClick": [Function],
      "title": "Select page (20 items)",
    },
    Object {
      "onClick": [Function],
      "title": "Select all (20 items)",
    },
  ],
  "onSelect": [Function],
  "toggleProps": Object {
    "children": Array [
      "0 selected",
    ],
  },
}
`;

exports[`useBulkSelectWithItems returns a allows to select/deselect page 2`] = `
Object {
  "checked": true,
  "isDisabled": false,
  "items": Array [
    Object {
      "onClick": [Function],
      "props": Object {
        "isDisabled": false,
      },
      "title": "Select none",
    },
    Object {
      "onClick": [Function],
      "title": "Unselect page (20 items)",
    },
    Object {
      "onClick": [Function],
      "title": "Unselect all (20 items)",
    },
  ],
  "onSelect": [Function],
  "toggleProps": Object {
    "children": Array [
      "20 selected",
    ],
  },
}
`;

exports[`useBulkSelectWithItems returns a allows to select/deselect page 3`] = `
Object {
  "checked": false,
  "isDisabled": false,
  "items": Array [
    Object {
      "onClick": [Function],
      "props": Object {
        "isDisabled": true,
      },
      "title": "Select none",
    },
    Object {
      "onClick": [Function],
      "title": "Select page (20 items)",
    },
    Object {
      "onClick": [Function],
      "title": "Select all (20 items)",
    },
  ],
  "onSelect": [Function],
  "toggleProps": Object {
    "children": Array [
      "0 selected",
    ],
  },
}
`;

exports[`useBulkSelectWithItems returns a bulk select configuration 1`] = `
Object {
  "all": Array [
    Object {
      "clearSelection": [Function],
      "selected": Array [],
      "selectedItems": Array [],
      "tableProps": Object {
        "canSelectAll": false,
        "onSelect": [Function],
      },
      "toolbarProps": Object {
        "bulkSelect": Object {
          "checked": false,
          "isDisabled": false,
          "items": Array [
            Object {
              "onClick": [Function],
              "props": Object {
                "isDisabled": true,
              },
              "title": "Select none",
            },
            Object {
              "onClick": [Function],
              "title": "Select page (20 items)",
            },
            Object {
              "onClick": [Function],
              "title": "Select all (20 items)",
            },
          ],
          "onSelect": [Function],
          "toggleProps": Object {
            "children": Array [
              "0 selected",
            ],
          },
        },
      },
      "transformer": [Function],
    },
    Object {
      "clearSelection": [Function],
      "selected": Array [],
      "selectedItems": Array [],
      "tableProps": Object {
        "canSelectAll": false,
        "onSelect": [Function],
      },
      "toolbarProps": Object {
        "bulkSelect": Object {
          "checked": false,
          "isDisabled": false,
          "items": Array [
            Object {
              "onClick": [Function],
              "props": Object {
                "isDisabled": true,
              },
              "title": "Select none",
            },
            Object {
              "onClick": [Function],
              "title": "Select page (20 items)",
            },
            Object {
              "onClick": [Function],
              "title": "Select all (20 items)",
            },
          ],
          "onSelect": [Function],
          "toggleProps": Object {
            "children": Array [
              "0 selected",
            ],
          },
        },
      },
      "transformer": [Function],
    },
  ],
  "current": Object {
    "clearSelection": [Function],
    "selected": Array [],
    "selectedItems": Array [],
    "tableProps": Object {
      "canSelectAll": false,
      "onSelect": [Function],
    },
    "toolbarProps": Object {
      "bulkSelect": Object {
        "checked": false,
        "isDisabled": false,
        "items": Array [
          Object {
            "onClick": [Function],
            "props": Object {
              "isDisabled": true,
            },
            "title": "Select none",
          },
          Object {
            "onClick": [Function],
            "title": "Select page (20 items)",
          },
          Object {
            "onClick": [Function],
            "title": "Select all (20 items)",
          },
        ],
        "onSelect": [Function],
        "toggleProps": Object {
          "children": Array [
            "0 selected",
          ],
        },
      },
    },
    "transformer": [Function],
  },
  "error": undefined,
}
`;

exports[`useBulkSelectWithItems returns respects filtered results 1`] = `
Object {
  "checked": false,
  "isDisabled": false,
  "items": Array [
    Object {
      "onClick": [Function],
      "props": Object {
        "isDisabled": true,
      },
      "title": "Select none",
    },
    Object {
      "onClick": [Function],
      "title": "Select page (5 items)",
    },
    Object {
      "onClick": [Function],
      "title": "Select all (5 items)",
    },
  ],
  "onSelect": [Function],
  "toggleProps": Object {
    "children": Array [
      "0 selected",
    ],
  },
}
`;

exports[`useBulkSelectWithItems returns respects filtered results 2`] = `
Object {
  "checked": true,
  "isDisabled": false,
  "items": Array [
    Object {
      "onClick": [Function],
      "props": Object {
        "isDisabled": false,
      },
      "title": "Select none",
    },
    Object {
      "onClick": [Function],
      "title": "Unselect page (5 items)",
    },
    Object {
      "onClick": [Function],
      "title": "Unselect all (5 items)",
    },
  ],
  "onSelect": [Function],
  "toggleProps": Object {
    "children": Array [
      "5 selected",
    ],
  },
}
`;
